components/GameScreen/GameScreen.tsx

```diff
@@ -1,21 +1,45 @@
-import { useState } from "react"
+import { useCallback, useEffect, useState, useMemo } from "react"
 import Image from "next/image"
-import _ from "lodash"
+import { Contract, ethers } from "ethers"
 import MintButton from "../MintButton"
 import { MUSIC_URLS } from "../../lib/consts"
+import abi from "../../lib/abi-musicGame.json"
+import getIpfsLink from "../../lib/getIpfsLink"
 
 interface IOption {
   id: string
   name: string
   imgUrl?: string
+  musicUrl?: string
 }
 interface IChecked {
   drums: boolean
   vocal: boolean
   bass: boolean
   guitar: boolean
 }
+
+type DecodedURI = {
+  name: string
+  image: string
+  animation_url: string
+  description: string
+  properties: {
+    name: string
+    number: number
+  }
+  tokenId: number
+}
+
+const getProvider = () => {
+  const goerliRpc = "https://ethereum-goerli-rpc.allthatnode.com"
```

Comment:
this should be ENV variable.

I will update in my PR, feel free to merge this and I will update 👍